stdlib/src/collections/list.mojo

```diff
@@ -416,6 +416,68 @@ struct List[T: CollectionElement](CollectionElement, Sized, Boolable):
             earlier_idx += 1
             later_idx -= 1
 
+    # TODO: Modify this to be regular method when issue 1876 is resolved
+    @staticmethod
+    fn index[
+        C: ComparableCollectionElement
+    ](
+        self: List[C], owned value: C, start: Int = 0, end: Optional[Int] = None
```

Comment:
This PR is amazing, thank you for the contribution @arvindavoudi and all the help @gabrieldemarmiesse.

One minor question we could possible address in a follow-up PR: Does this `value` argument used for comparision need to be `owned`, or could use the default `borrowed` argument convention? I.e.:

```mojo
self: List[C], value: C, ....
```

Logically I think `borrowed` would make sense, since the element isn't mutated or consumed by this method.

Is there some other reason this methods needs its own unique copy of the `value` argument?

Reply:
I had the same thought this morning — something we should follow-up on.

Reply:
No worries :) And yep, we can take of this internally as well.

Once we sync our internal changes out again, it will be easier for a community member to make this change as well if they wanted to. Its a small fix though, so I might just do it real quick internally if no one else gets to it first :)

Reply:
This has been fixed by our new intern @Dan13llljws in [this commit here](https://github.com/modularml/mojo/commit/f2f9e77ecf9780af4ecdf28ca170ca9276f586a9) — welcome to the team @Dan13llljws, and congratulations on your first Mojo🔥 contribution! 🎉  😄